src/compiler/context.js

```diff
@@ -0,0 +1,228 @@
+          }
+          ${afterLoop}
+        } else {
+          ${notArrayAction}
+        }
+      }.bind(this))`;
+
+      if (hasElseBody) {
+        output += `.catch(function(err) {
+          ${elseBodyAction};
+        }.bind(this))`;
+      }
+
+      if (isInHtmlAttribute) {
+        return output;
+      } else {
+        this.append(null, `      ${util.createPlaceholder(this)};\n`, null);
+        this.append(null, null, `      var on${indexHash} = td.${util.getTdMethodName('getNodeAtIdxPath')}(root, ${JSON.stringify(indexes)});
+      ${output};\n`);
+      }
+    },
+
+    helper(node) {
+      let maxTdIndex = this.tdBodies.length - 1;
+      let tdIndex = tornadoBodiesPointer;
+      let indexes = this.htmlBodies[tdIndex].htmlBodiesIndexes;
+      let indexHash = indexes.join('');
+      let paramsHash, bodiesHash;
+      paramsHash = node.params.reduce((acc, param) => {
+        let paramVal = param.val;
+        if (Array.isArray(paramVal)) {
+          paramVal = `td.${util.getTdMethodName('get')}(c, ${JSON.stringify(paramVal[1].key)})`;
+        } else {
+          paramVal = `'${paramVal}'`;
+        }
+        acc.push(`${param.key}: ${paramVal}`);
+        return acc;
+      }, []);
+      paramsHash = `{${paramsHash.join(',')}}`;
+      bodiesHash = node.bodies.reduce((acc, body, idx) => {
+        let bodyName = body[1].name;
+        acc.push(`${bodyName}: this.r${maxTdIndex + idx + 2}.bind(this)`);
+        return acc;
+      }, []);
+      if (node.body.length) {
+        bodiesHash.push(`main: this.r${maxTdIndex + 1}.bind(this)`);
+      }
+      bodiesHash = `{${bodiesHash.join(',')}}`;
+      if (this.state !== STATES.HTML_ATTRIBUTE) {
+        this.append(null, `      ${util.createPlaceholder(this)};\n`,
+                          `      var on${indexHash} = td.${util.getTdMethodName('getNodeAtIdxPath')}(root, ${JSON.stringify(indexes)});
+        td.${util.getTdMethodName('helper')}('${node.key.join('')}', c, ${paramsHash}, ${bodiesHash}).then(function(val) {
+          td.${util.getTdMethodName('replaceNode')}(on${indexHash}, val);
+        });\n`);
+      }
+    },
+
+    block(node) {
+      let tdIndex = tornadoBodiesPointer;
+      let indexes = this.htmlBodies[tdIndex].htmlBodiesIndexes;
+      let indexHash = indexes.join('');
+      if (this.state !== STATES.HTML_ATTRIBUTE) {
+        this.append(null, `      ${util.createPlaceholder(this)};\n`, null);
+        this.append(null, null, `      var on${indexHash} = td.${util.getTdMethodName('getNodeAtIdxPath')}(root, ${JSON.stringify(indexes)});
+      td.${util.getTdMethodName('replaceNode')}(on${indexHash}, td.${util.getTdMethodName('block')}('${node.blockName}', ${node.blockIndex}, c, this));\n`);
+      } else {
+        return `td.${util.getTdMethodName('nodeToString')}(td.${util.getTdMethodName('block')}('${node.blockName}', ${node.blockIndex}, c, this))`;
+      }
+    },
+
+    inlinePartial(node) {
+      console.log(node);
```

Comment:
lol `eslint`. for some reason i thought it better to console log rather than remove the param.